src/lib/min-max-zoom/__test__/min-max-zoom.test.js

```diff
@@ -0,0 +1,255 @@
+import { getExtraneousMinMaxZoom } from '../min-max-zoom';
+
+describe('getExtraneousMinMaxZoom', () => {
+  describe('unnecessary min and max zooms', () => {
+    let style = {
+      id: 'my-style',
+      layers: [
+        {
+          id: 'layer-1',
+          paint: {
+            'fill-color': 'red'
+          }
+        }
+      ]
+    };
+    test('minzoom redundant', () => {
+      const testStyle = JSON.parse(JSON.stringify(style));
+      testStyle.layers[0].minzoom = 0;
+      const actual = getExtraneousMinMaxZoom(testStyle);
+      const expected = ['layer-1: contains unnecessary minzoom'];
+
+      expect(actual).toEqual(expected);
+    });
+
+    test('maxzoom redundant', () => {
+      const testStyle = JSON.parse(JSON.stringify(style));
+      testStyle.layers[0].maxzoom = 24;
+      const actual = getExtraneousMinMaxZoom(testStyle);
+      const expected = ['layer-1: contains unnecessary maxzoom'];
+
+      expect(actual).toEqual(expected);
+    });
+  });
+
+  describe('scale expression zooms outside min and max', () => {
+    let style = {
+      id: 'my-style',
+      layers: [
+        {
+          id: 'layer-1',
+          paint: {
+            'fill-color': [
+              'interpolate',
+              ['linear'],
+              ['zoom'],
+              5,
+              'red',
+              10,
+              'blue',
+              15,
+              'green'
+            ]
+          }
+        }
+      ]
+    };
+
+    test('simple interpolate past maxzoom warnings', () => {
+      const testStyle = JSON.parse(JSON.stringify(style));
+      testStyle.layers[0].maxzoom = 9;
+
+      const actual = getExtraneousMinMaxZoom(testStyle);
+      const expected = [
+        'layer-1: fill-color expression contains zooms greater than the maxzoom - 10, 15'
```

Comment:
I appreciate that these are written to be consistent with other warnings but I find them a little difficult to read, specifically I'm not sure what `10` and `15` are. I wonder if something like this would be more clear?

```suggestion
        'layer-1: fill-color expression contains zooms greater than the maxzoom - found zooms 10, 15 but maxzoom is 9'
```

Reply:
I was skeptical before, but I ended up removing this warning all together just now. We tried this in practice and it's generally not pragmatic. There's just too many reasons someone might write an expression this way and there's not a real performance impact in changing it.